src/utils/decoder_sql/decoder_sql.c

```diff
@@ -0,0 +1,704 @@
+
+void
+_PG_init(void)
+{
+	/* other plugins can perform things here */
+}
+
+/* specify output plugin callbacks */
+void
+_PG_output_plugin_init(OutputPluginCallbacks *cb)
+{
+	AssertVariableIsOfType(&_PG_output_plugin_init, LogicalOutputPluginInit);
+
+	cb->startup_cb = decoder_sql_startup;
+	cb->begin_cb = decoder_sql_begin_txn;
+	cb->change_cb = decoder_sql_change;
+	cb->commit_cb = decoder_sql_commit_txn;
+	cb->shutdown_cb = decoder_sql_shutdown;
+}
+
+/* initialize this plugin */
+static void
+decoder_sql_startup(LogicalDecodingContext *ctx, OutputPluginOptions *opt, bool is_init)
+{
+	ListCell *option;
+	DecoderRawData *data;
+
+	data = palloc(sizeof(DecoderRawData));
+	data->context =
+		AllocSetContextCreate(ctx->context, "Raw decoder context", ALLOCSET_DEFAULT_SIZES);
+	data->include_transaction = false;
+	data->include_relations = NIL;
+	data->target_hypertable = NULL;
+
+	ctx->output_plugin_private = data;
+
+	/* Default output format */
+	opt->output_type = OUTPUT_PLUGIN_TEXTUAL_OUTPUT;
+
+	foreach (option, ctx->output_plugin_options)
+	{
+		DefElem *elem = lfirst(option);
+
+		Assert(elem->arg == NULL || IsA(elem->arg, String));
+
+		if (strcmp(elem->defname, "include_transaction") == 0)
+		{
+			/* if option does not provide a value, it means its value is true */
+			if (elem->arg == NULL)
+				data->include_transaction = true;
+			else if (!parse_bool(strVal(elem->arg), &data->include_transaction))
+				ereport(ERROR,
+						(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+						 errmsg("could not parse value \"%s\" for parameter \"%s\"",
+								strVal(elem->arg),
+								elem->defname)));
+		}
+		else if (strcmp(elem->defname, "output_format") == 0)
+		{
+			char *format = NULL;
+
+			if (elem->arg == NULL)
+				ereport(ERROR,
+						(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+						 errmsg("No value specified for parameter \"%s\"", elem->defname)));
+
+			format = strVal(elem->arg);
+
+			if (strcmp(format, "textual") == 0)
+				opt->output_type = OUTPUT_PLUGIN_TEXTUAL_OUTPUT;
+			else if (strcmp(format, "binary") == 0)
+				opt->output_type = OUTPUT_PLUGIN_BINARY_OUTPUT;
+			else
+				ereport(ERROR,
+						(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+						 errmsg("Incorrect value \"%s\" for parameter \"%s\"",
+								format,
+								elem->defname)));
+		}
+		else if (strcmp(elem->defname, "include_relations") == 0)
+		{
+			char *relargs = NULL;
+			List *namelist = NIL;
+			RangeVar *relvar;
+			Relation rel;
+			List *relname_list;
+			ListCell *l;
+			MemoryContext old = MemoryContextSwitchTo(ctx->context);
+
+			if (elem->arg == NULL)
+				ereport(ERROR,
+						(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+						 errmsg("No value specified for parameter \"%s\"", elem->defname)));
+
+			/* Need a modifiable copy of string */
+			relargs = pstrdup(strVal(elem->arg));
+
+			/* Parse string into list of identifiers */
+			if (!SplitIdentifierString(relargs, ',', &namelist) || namelist == NIL)
```

Comment:
instead of splitting by `,` and other stuff - wouldn't it be better to simply support that this `include_relations` might be present multiple times?